src/test/java/teammates/test/util/TestUrl.java

```diff
@@ -0,0 +1,16 @@
+package teammates.test.util;
+
+import teammates.common.util.AppUrl;
+import teammates.test.driver.TestProperties;
+
+/**
+ * An extension of {@link AppUrl}, differing in the base URL used for conversion.
+ * In this class, the base URL used is the value of test.app.url in test.properties.
+ */
+public class TestUrl extends AppUrl {
+
+    public TestUrl(String url) {
+        super(TestProperties.inst().TEAMMATES_URL, url);
+    }
+    
+}
```

Comment:
I have a feeling we are misusing inheritance here, but can't think of a better solution at the moment.
May be instead of a TestUrl subclass, there should be a factory method called getAppUrlForTestingApp(...) which returns an AppUrl objects with the correct base Url?